src/08_ecosystem/00_chapter.md

```diff
@@ -0,0 +1,63 @@
+# The Async Ecosystem
+Most Rust asynchronous projects will need crates provided by the community to function.
+The [`futures` crate](https://docs.rs/futures/) contains traits and functions useful for writing async code.
+In addition, there are several crates that provide async runtimes,
+which are used to run asynchronous functions from synchronous code.
+This section introduces the crates most commonly used in async projects.
+
+## Asynchronous Runtimes
+Async runtimes are libraries that allow async functions to be run from sync functions, including the `main` function.
+Runtimes usually bundle together a *reactor* with one or more *executors*.
+Reactors provide subscription mechanisms for external events, like async I/O, interprocess communication, and timers.
+In an async runtime, subscribers are typically futures representing low-level I/O operations.
+Executors handle the scheduling and execution of tasks.
+They keep track of running and suspended tasks, poll futures to completion, and wake tasks when they can make progress.
+The word "executor" is frequently used interchangeably with "runtime".
+
+## Choosing a Runtime
+There is no asynchronous runtime in the standard library, and none are officially recommended.
+The following crates provide popular runtimes.
+
+- [futures](https://docs.rs/futures/): The `futures` crate is not an async ecosystem like other crates in this list,
+but it provides useful traits like `Stream`, `Sink`, `AsyncRead`, and `AsyncWrite`, and utilities such as combinators.
+These utilities and traits may eventually become part of the standard library.
+`futures` has an executor, `futures-executor`, but a common choice is to use utilities from `futures` with an
+executor from another crate.
+- [Tokio](https://docs.rs/tokio/): A popular async ecosystem with HTTP, gRPC, and tracing frameworks.
+- [async-std](https://docs.rs/async-std/): A crate that provides asynchronous counterparts to standard library components.
+- [smol](https://docs.rs/smol/): A small, simplified async runtime.
+Provides the `Async` trait that can be used to wrap structs like `UnixStream` or `TcpListener`.
+- [fuchsia-async](https://fuchsia.googlesource.com/fuchsia/+/master/src/lib/fuchsia-async/):
+An executor for use in the Fuchsia OS.
+
+Not all async runtimes are compatible with each other.
+Notably, `Tokio` uses the `mio` reactor and defines its own versions of async I/O traits,
+including `AsyncRead` and `AsyncWrite`.
+On its own, it's not compatible with `async-std` and `smol`,
+which rely on the [`async-executor` crate](https://docs.rs/async-executor).
+The [`async_compat` crate](https://docs.rs/async_compat) can be used for compatibility between `Tokio` and other runtimes.
+
+More generally, futures that interact with async I/O, interprocess communication, or timers are tied
```

Comment:
I suggest again a different h2 subheading, like "Determining ecosystem compatibility"

Although many frameworks and libraries impose requirements on specific async ecosystems, a large set of async code can be used with any ecosystem. Ecosystem constraints are not always documented, so how can you tell whether a library, type or function depends on an ecosystem or not? As a rule of thumb:

- Any async code that interacts with _async IO, timers, IPC or tasks_ generally depends on a specific async executor or reactor.
- All other async code, such as async expressions, combinators, synchronization types, and streams are usually ecosystem independent, provided that any nested futures are also ecosystem independent.
- Non-send (`!Send`) futures requires that a runtime that supports local tasks.